### ISSUE TITLE
CairoMakie fails to save image with interpolate=false to pdf

### ISSUE THREAD

**Issue by Sleort** [MakieOrg/Makie.jl#2514]
```julia
using CairoMakie, FileIO

img = load(assetpath("cow.png"))

# With interpolation works when saving to pdf:
f = image(img; interpolate=true)    
save("test.pdf", f)   #ok

# Without interpolation fails when saving to png:
f = image(img; interpolate=false)    
save("test.png", f)   #ok

# Without interpolation fails when saving to pdf:
f = image(img; interpolate=false)    
save("test.pdf", f)   #errors
```
where the error message reads
```julia
ERROR: Error in conversion pipeline. xs and ys should have size ni+1, nj+1. Found: xs: 348, ys: 418, ni: 348, nj: 418
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] draw_atomic(scene::Scene, screen::CairoMakie.Screen{CairoMakie.PDF}, primitive::Union{Heatmap, Image})
    @ CairoMakie ~/.julia/packages/CairoMakie/ihglX/src/primitives.jl:629
  [3] draw_plot(scene::Scene, screen::CairoMakie.Screen{CairoMakie.PDF}, primitive::Image{Tuple{StepRangeLen{Float32, Float64, Float64, Int64}, StepRangeLen{Float32, Float64, Float64, Int64}, Matrix{ColorTypes.RGBA{Float32}}}})
    @ CairoMakie ~/.julia/packages/CairoMakie/ihglX/src/infrastructure.jl:106
  [4] cairo_draw(screen::CairoMakie.Screen{CairoMakie.PDF}, scene::Scene)
    @ CairoMakie ~/.julia/packages/CairoMakie/ihglX/src/infrastructure.jl:47
  [5] backend_show(screen::CairoMakie.Screen{CairoMakie.PDF}, io::IOStream, #unused#::MIME{Symbol("application/pdf")}, scene::Scene)
    @ CairoMakie ~/.julia/packages/CairoMakie/ihglX/src/display.jl:97
  [6] (::Makie.var"#1003#1004"{Module, Bool, Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, Makie.FigureAxisPlot, MIME{Symbol("application/pdf")}, Scene})(io::IOStream)
    @ Makie ~/.julia/packages/Makie/Ggejq/src/display.jl:261
  [7] open(::Makie.var"#1003#1004"{Module, Bool, Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, Makie.FigureAxisPlot, MIME{Symbol("application/pdf")}, Scene}, ::String, ::Vararg{String}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./io.jl:384
  [8] open
    @ ./io.jl:381 [inlined]
  [9] save(file::File{DataFormat{:PDF}, String}, fig::Makie.FigureAxisPlot; resolution::Tuple{Int64, Int64}, backend::Module, update::Bool, screen_config::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/Ggejq/src/display.jl:256
 [10] save(file::File{DataFormat{:PDF}, String}, fig::Makie.FigureAxisPlot)
    @ Makie ~/.julia/packages/Makie/Ggejq/src/display.jl:234
 [11] save(filename::String, fig::Makie.FigureAxisPlot; args::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/Ggejq/src/display.jl:231
 [12] save(filename::String, fig::Makie.FigureAxisPlot)
    @ Makie ~/.julia/packages/Makie/Ggejq/src/display.jl:228
 [13] top-level scope
    @ REPL[17]:1
```

This is with `CairoMakie v0.10.0`

**Comment by kapple19** [comment]
I have the same issue saving to an SVG file with CairoMakie version 0.11.9.

**Comment by asinghvi17** [comment]
I guess the vectors of `xs` and 	`ys` aren't being converted to edges in this case?

**Comment by kapple19** [comment]
I started looking at this issue today, but on today's master branch, the MWE in @Sleort's initial post runs without error.
My use case also runs without error for CairoMakie v0.11.10.

**Comment by Sleort** [comment]
Yeah, it looks like this has been resolved (at least on CairoMakie v0.11.10). :partying_face:  I'll close the issue.
